src/app/shell/personal-cabinet/parent/create-child/create-child.component.html

```diff
@@ -14,9 +14,12 @@ <h3 class="wrapper-title">{{(editMode) ? 'РЕДАГУВАТИ' : 'ДОДАТИ'
     </ng-container>
 
     <div *ngIf="!editMode" fxLayout="column" fxLayoutAlign="center center" class="addChild">
-      <button class="btn-add" mat-button (click)="addChild()" [disabled]="ChildrenFormArray.invalid || ChildrenFormArray.length > 21">
+      <button *ngIf="ChildrenFormArray.controls.length < 20 - (childrenCards$ | async)?.totalAmount; else noMoreChildren" class="btn-add" mat-button (click)="addChild()" [disabled]="ChildrenFormArray.invalid">
```

Comment:
I think (childrenCards$ | async)?.totalAmount;
? is redundant, because as I remember we receive an object where entities and total amount
so at least it is going to be total amount 0 and entities []

Reply:
+

Reply:
try mock this value for test

Reply:
You can leave it like that as well, but you can check some tests how to fix it. You need to set for test this value, so it won't fail
like from the store you received undefined but it is not possible in reality